### PR TITLE
Fix Network Server field mask validation and Console usage

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

References https://github.com/TheThingsNetwork/lorawan-stack/pull/3437

#### Changes

- Removed `recent_uplinks` and `recent_downlinks` from the NS end device registry field mask validation. The fields are no longer used post #3437 
- Replaced the said fields with their `mac_state` and `pending_mac_state` variants in the field mask validation
- Console now uses `mac_state.recent_uplinks` for recent uplink status instead

#### Testing

I've reproduced this by trying to access a device in the Console. So did just that.

##### Regressions

The removed fields were already unavailable after #3437, so this just makes validation catch up to the registry behavior. No other regressions are expected.

#### Notes for Reviewers

I didn't add the fields to the `Set` RPC since they weren't there before. We can relax that later, should it be needed.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/ttnpb/field_mask_validation.go [hunk 1]
```diff
@@ -313,6 +313,8 @@ var nsEndDeviceReadFieldPaths = [...]string{
 	"mac_state.queued_join_accept.request.selected_mac_version",
 	"mac_state.queued_responses",
 	"mac_state.rx_windows_available",
+	"mac_state.recent_downlinks",
```
Comment (rvolosatovs) [inline pkg/ttnpb/field_mask_validation.go:315]: please make sure this is sorted alphabetically

File: pkg/ttnpb/field_mask_validation.go [hunk 2]
```diff
@@ -515,6 +517,8 @@ var nsEndDeviceReadFieldPaths = [...]string{
 	"pending_mac_state.queued_join_accept.request.selected_mac_version",
 	"pending_mac_state.queued_responses",
 	"pending_mac_state.rx_windows_available",
+	"pending_mac_state.recent_downlinks",
```
Comment (rvolosatovs) [inline pkg/ttnpb/field_mask_validation.go:519]: IDEM

**Review by rvolosatovs — APPROVED** [review]

File: pkg/ttnpb/field_mask_validation.go (hunk 1 above)
Comment (adriansmares) [reply pkg/ttnpb/field_mask_validation.go:315]: Should I sort only `mac_state` lines only, or the whole slice ? I wanted to do this initially but the diff was large so I thought I'm missing something about the sorting.

Comment (rvolosatovs) [reply pkg/ttnpb/field_mask_validation.go:315]: The whole slice is supposed to be sorted and if it's not - we should fix that

Comment (rvolosatovs) [reply pkg/ttnpb/field_mask_validation.go:315]: Note the case-insensitive sorting!

Comment (rvolosatovs) [reply pkg/ttnpb/field_mask_validation.go:315]: In `vim`: `:sort i`

**Review by kschiffer — APPROVED** [review]
Codeowned files LGTM.

**Review by johanstokking — APPROVED** [review]

**Review by rvolosatovs — APPROVED** [review]

**Review by htdvisser — APPROVED** [review]
Nothing to add for `.go` files.

**Pull request merged**